File: GFG Weekly Coding Contests/GFG Weekly Coding Contest - 133/Maximum Happiness/Solution.java
import java.util.*;

class Solution {
    public static int maxHappy(int n, int[] bags, int m, int[] req) {
        // code here
        Arrays.sort(bags);
        Arrays.sort(req);
        
        int i = 0, j = 0, ans = 0;
        
        while(i<n && j<m) {
            if(bags[i] >= req[j]) {
                i++;
                j++;
                ans++;
            }
            else i++;
        }
        
        return ans;
    }
}

File: GFG Weekly Coding Contests/GFG Weekly Coding Contest - 133/Maximum Happiness/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
  public:
    int maxHappy(int n, vector<int> &bags, int m, vector<int> &req) {
        sort(bags.begin(), bags.end());
        sort(req.begin(), req.end());
        int ans = 0, i=0, j=0;
        
        while(i<n && j<m){
            if(bags[i] >= req[j]){
                i++;
                j++;
                ans++;
            }
            else if(bags[i] < req[j]){
                i++;
            }
        }
        return ans;
    }
};